8 packages from gitlab.com/nomadic-labs/resto/-/archive/v0.2/resto-v0.2.tar.gz

raphael-proust opened the pull request:
This pull-request concerns:
-`ezresto.0.2`: A minimal OCaml library for type-safe HTTP/JSON RPCs
-`ezresto-directory.0.2`: A minimal OCaml library for type-safe HTTP/JSON RPCs
-`resto.0.2`: A minimal OCaml library for type-safe HTTP/JSON RPCs
-`resto-cohttp.0.2`: A minimal OCaml library for type-safe HTTP/JSON RPCs
-`resto-cohttp-client.0.2`: A minimal OCaml library for type-safe HTTP/JSON RPCs
-`resto-cohttp-server.0.2`: A minimal OCaml library for type-safe HTTP/JSON RPCs - server library
-`resto-directory.0.2`: A minimal OCaml library for type-safe HTTP/JSON RPCs
-`resto-json.0.2`: A minimal OCaml library for type-safe HTTP/JSON RPCs



---
* Homepage: https://gitlab.com/nomadic-labs/resto
* Source repo: git+https://gitlab.com/nomadic-labs/resto
* Bug tracker: https://gitlab.com/nomadic-labs/resto/issues

---
:camel: Pull-request generated by opam-publish v2.0.0

raphael-proust commented:
I'm not sure how to do this release considering the source tree declares several packages in the same directory.

Lwt me know if I need to change anything.

kit-ty-kate commented on packages/ezresto-directory/ezresto-directory.0.2/opam:
```diff
@@ -0,0 +1,29 @@
+depends: [
+  "ocaml" { >= "4.07" }
+  "dune" { >= "1.7" }
+  "ezresto" {= "0.2" }
```
For next time be aware you can use ` {= version}` instead of `{= "0.2"}`

raphael-proust replied:
Oh nice! I didn't know that trick!

kit-ty-kate commented on packages/ezresto/ezresto.0.2/opam:
```diff
@@ -0,0 +1,31 @@
+
+build: [
+  [ "dune" "build" "-p" name "-j" jobs ]
+  [ "dune" "build" "@runtest_ezresto" "-j" jobs ] { with-test }
```
```suggestion
  [ "dune" "build" "-p" name "-j" jobs ] { with-test }
```
I think this should work in your case as well. If it does not please use the `--profile=release` argument as the default profile might break packages in the future

raphael-proust replied:
I tried your suggested changes but it doesn't work. I think you meant `runtest` instead of `build`?
But that also gives some trouble probably because the tests for one package depend on another and it confuses opam/dune.

I'll ship it without the tests in the opam packages. The tests are run on the CI anyway and I'll consider that to be enough for this release.

kit-ty-kate commented on packages/resto/resto.0.2/opam:
```diff
@@ -0,0 +1,31 @@
+
+build: [
+  [ "dune" "build" "-p" name "-j" jobs ]
+  [ "dune" "build" "@runtest_resto" "-j" jobs ] { with-test }
```
```suggestion
  [ "dune" "build" "-p" name "-j" jobs ] { with-test }
```
same here

kit-ty-kate commented:
Otherwise releasing several packages at once is pretty common and totally ok, don't worry

raphael-proust commented:
Thanks for the comments! I fixed them all (although the test part I fixed by simply removing them which I think is ok because we run the test as part of the CI).

kit-ty-kate commented:
Thanks!
